README.md

```diff
@@ -1600,6 +1604,18 @@ $ helm --upgrade install actions-runner-controller/actions-runner-controller \
   admissionWebHooks.caBundle=${CA_BUNDLE}
 ```
 
+#### Using helm to generate CA and certificates
+
+Set the Helm chart values as follows:
+
+```shell
+$ CA_BUNDLE=$(cat path/to/ca.pem | base64)
+$ helm --upgrade install actions-runner-controller/actions-runner-controller \
+  certManagerEnabled=false
+```
+
+This generates a temporary CA using the helm `genCA` function and issues a certificate for the webhook. Note that this approach rotates the CA and certificate each time `helm install` or `helm upgrade` are run.
```

Comment:
Does this imply that there might be some admission webhook downtime each time you run `helm upgrade`, because it would take some time until all the ARC processes and ARC pods catches up the updated serving-cert secret, where the caBundle included in the admissionwebhook is updated without any delay?

Reply:
Yes, that is what is going to happen (see https://github.com/helm/helm/issues/10731), until the pods refresh the certificate, the kubernetes apiserver will get a TLS error calling the admissions web hook. It is not ideal but this is a solution for when you don't want to run cert-manager

An option would be to move this to a helm pre-start hook but I have an issue with the way code is laid out in the chart because you cannot share the cert or ca objects between different helm files so the hook would need to contain both the certificate config map as well as the mutating web hook configuration.

Other projects like `kyverno` use the CA secret to generate the web hook definitions from the application (see https://github.com/kyverno/kyverno/blob/42a2df56c17af311a7c006df1532d26f58347547/pkg/webhookconfig/registration.go#L241) but with the ARC doing it from the helm chart it is difficult to properly correlate this dependency between the certificate and the mutatingwebhookconfiguration.

Reply:
> Yes, that is what is going to happen

Gotcha!

I think you've already covered it in the new README section and given it's documented I think it's fine.

> An option would be to move this to a helm pre-start hook

Looks very hard indeed! Thanks for sharing your insight here.

> but with the ARC doing it from the helm chart it is difficult to properly correlate this dependency between the certificate and the mutatingwebhookconfiguration.

Understood. If we were to follow a path similar to kyverno, we should redesign ARC to create/update the mutatingwebhookconfigs by itself, not by helm.

kyverono does seem to have it's own certmanager (not cert-manager :) ) controller which probably depends on the user provided root CA to generate certs. Doing same on ARC would enable us to potentially drop the requirement on cert-manager, simplifying the deployment of ARC. Are you interested in it?